src/coreclr/nativeaot/Runtime/amd64/WriteBarriers.S

```diff
@@ -307,7 +306,7 @@ LOCAL_LABEL(RhpByRefAssignRef_CheckCardTable):
     shr     rcx, 0x0B
     mov     r10, [C_VAR(g_card_table)]
     cmp     byte ptr [rcx + r10], 0x0FF
-    je      LOCAL_LABEL(RhpByRefAssignRef_NotInHeap)
+    je      LOCAL_LABEL(RhpByRefAssignRef_NoBarrierRequired)
 
```

Comment:
CoreCLR versions of these write-barriers use pattern like this:

```
        test    byte ptr [rcx + r10], al
        je      SetCardTableBit_ByRefWriteBarrier
        REPRET

    SetCardTableBit_ByRefWriteBarrier:
```

It makes the hot path of the write barrier fit into fewer cache lines and it makes the jump better statically predicted (forward conditional branches are predicted not taken by default).